12-offer-encoding.md

```diff
@@ -360,83 +405,88 @@ invoices is `lnr`.
 
 ## Requirements for Invoice Requests
 
-The writer of an invoice_request:
-  - SHOULD set `payer_info` to an unpredictable series of bytes.
-  - MUST set `payer_key` to a transient public key.
-  - MUST remember the secret key corresponding to `payer_key`.
-  - MUST set `offer_id` to the Merkle root of the offer as described in [Signature Calculation](#signature-calculation).
-  - MUST NOT set or imply any `chain_hash` not set or implied by the offer.
-  - MUST set `signature` `sig` as detailed in [Signature Calculation](#signature-calculation) using the `payer_key`.
-  - if the offer had a `quantity_min` or `quantity_max` field:
-    - MUST set `quantity`
+The writer:
+  - MUST copy all fields from the offer (including unknown fields).
```

Comment:
I didn't quite follow the conversation in the spec meeting. What's the advantage of copying all the offer TLVs into the invoice request vs having a single `offer_tlvs` type which contains the full offer tlv stream (tlv-in-a-tlv)?

Reply:
You can't partially prove that, you'd need to expose all of it.  And we were reflecting *many* fields already, so this is actually simpler.

Reply:
I suppose reflecting the fields is functionally equivalent to parsing the two (or more) tlv streams back-to-back over a specific type rane? (modulo `invoice_request_payer_info` which may be can be thought of as its own TLV stream?)

Reply:
But simpler.  If you have two TLVs you need a wrapper format (they don't naturally separate, in general).  And you have, again, the "but we want a single proof" problem, which now has to handle two TLVs.